src/mon/MDSMonitor.cc

```diff
@@ -203,7 +203,7 @@ void MDSMonitor::encode_pending(MonitorDBStore::TransactionRef t)
     }
     for (const auto &metric : health.metrics) {
       int const rank = info.rank;
-      health_check_t *check = &new_checks.add(
+      health_check_t *check = &new_checks.get_or_add(
```

Comment:
yeah, this happens. and should in the case of multi-mds deployment. but if we always overwrite the `health.metrics` from one mds instance with those from another. why bothering iterating through the fsmap for all mds instances? 

in the multi-mds setting, if all MDSes are in sync in the sense that the health data are same across all active MDSes, we can just collect the health check info from one MDS then.

if they are not in sync, we probably need to find out which copy is authoritative.

but either way, i am just not sure we should do the overwrite.

@jcsp @ukernel what do you think?